event/dispatcher_test.go

```
@@ -0,0 +1,106 @@
+	dispatcher := NewDispatcher()
+
+	t.Run("Multiple event bindings based on callback functions", func(t *testing.T) {
+		dispatcher.Reset()
+
+		dispatcher.RegisterListeners(
+			AdaptListener(ListenerFunc[*UserEvent](func(ctx context.Context, event *UserEvent) error {
+				t.Logf("this is user listener, the name is %s", event.Name)
+				assert.Equal(t, "zhangsan", event.Name)
+				return nil
+			})),
+			AdaptListener(ListenerFunc[*OrderEvent](func(ctx context.Context, event *OrderEvent) error {
```

golangci-lint (1.62.2)

[warning] unused-parameter: parameter 'ctx' seems to be unused, consider removing or renaming it as _

(revive)